lyrics-fetcher-neteasecloud.el

```diff
@@ -0,0 +1,210 @@
+  "Perform a lyrics search on 'music.163.com'.
+
+The flow is as follows:
+1. Send a POST /search request with a text query
+2. Pick the first result (or prompt user if SYNC is non-nil)
+3. Fetch lyrics
+4. Call CALLBACK with the resulting lyrics string
+
+TRACK should be EMMS-compatible alist or string, take a look at
+`lyrics-fetcher-neteasecloud--format-query'.  If the search is
+successful, CALLBACK will be called with the result.
+
+If SYNC is non-nil, perform request synchronously and ask the
+user to pick the matching search result.
+
+When EDIT is non-nil, edit the query in minibuffer before search.
+Genius usually struggles to find song if there is extra
+information in the title."
+  (lyrics-fetcher-neteasecloud--do-query
+   track
+   (lambda (data)
+     (lyrics-fetcher-neteasecloud--fetch-lyrics
+      (lyrics-fetcher-neteasecloud--get-song-id data sync)
+      callback
+      sync))
+   sync
+   edit))
+
+(defun lyrics-fetcher-neteasecloud--fetch-lyrics (song-id callback &optional sync)
+  "Fetch lyrics from 'music.163.com' page at URL and call CALLBACK with the result.
+
+SONG-ID is a sequence of number which indicates a song, it can be
+returned by 'lyrics-fetcher-neteasecloud--get-song-id' If SYNC is
+non-nil, the request will be performed synchronously."
+  (message "Getting lyrics from NeteaseCloud API...")
+  (request
+    (format "http://music.163.com/api/song/lyric?id=%s&lv=1&kv=1&tv=-1" song-id)
+    :parser 'json-read
+    :sync sync
+    :success (cl-function
+     (lambda (&key data &allow-other-keys)
+       (funcall callback (alist-get 'lyric (alist-get 'lrc data)))))
+    :error
+    (cl-function
+     (lambda (&key error-thrown &allow-other-keys)
+       (message "Error!: %S" error-thrown)))))
+
+(defun lyrics-fetcher-neteasecloud--do-query (track callback &optional sync edit)
+  "Perform a song search on 'music.163.com'.
+
+TRACK should be EMMS-compatible alist or string, take a look at
+`lyrics-fetcher-neteasecloud--format-query'.  If the search is
+successful, CALLBACK will be called with the result.
+
+SYNC determines whether the request is synchronous.  The parameter
+is useful when it is necessary to ask the user for something right
+after the request.
+
+When EDIT is non-nil, edit the query in minibuffer before search."
+  (message "Sending a query to NeteaseCloud API...")
+  (request "http://music.163.com/api/search/get/"
+    :type "POST"
+    :data `(("s" . ,(lyrics-fetcher-neteasecloud--maybe-edit-query
+		     (lyrics-fetcher-neteasecloud--format-query track)
+		     edit))
+	    ("limit" . "10")
+	    ("type" . "1")
+	    ("offset" . "0"))
+    :parser 'json-read
+    :sync sync
+    :success (cl-function
+              (lambda (&key data &allow-other-keys)
+                (funcall callback data)))
+    :error (cl-function
+            (lambda (&key error-thrown &allow-other-keys)
+              (message "Error!: %S" error-thrown)))
+    ))
```

Comment:
Please, remove this trailing parenthesis as well. And the rest and further through the file